lib/targets/cuda/malloc.cpp

```diff
@@ -679,6 +840,22 @@ namespace quda
       }
     }
 
+#ifdef NVSHMEM_COMMS
+#define shmem_free(ptr) quda::shmem_free_(__func__, quda::file_name(__FILE__), __LINE__, ptr)
+    void flush_shmem()
+    {
+      if (shmem_memory_pool) {
+        std::multimap<size_t, void *>::iterator it;
```

Comment:
Since `multimap`s can have multiple keys pointing to the same value, shouldn't we need some guarantee that a value hasn't already been `shmem_free`'d? Alternatively, does the shmem memory pool actually need to be a `multimap` (I think it's unique among the pools as being one), or would a `map` suffice?

Reply:
This is the same thing we use for the other pools as well. But it actually brings up the question whether we do need a pool for symmetric allocations. Since we move to hide all shmem allocations behind `device_comms_pinned_` which falls back to `device_pinned_` if shmem is not used maybe just using a pure shmem_malloc is sufficient here.

Reply:
no longer there